doc: update quick start for rc.1 release (#100)
notaryproject/notaryproject.dev: content/en/docs/quickstart.md

Create and run an OCI-compatible registry on your development computer using Docker and the [oras-project/registry](https://github.com/oras-project/distribution/pkgs/container/registry) container image. The following command creates a registry that is accessible at `localhost:5000`.

```console
docker run -d -p 5000:5000 ghcr.io/oras-project/registry:v1.0.0-rc.3
```

> NOTE:

> > Since v1.0.0-rc.1 release, by default, Notation stores signatures using OCI artifact Manifest, which is defined in [OCI Image spec v1.1.0](https://github.com/opencontainers/image-spec/tree/v1.1.0-rc2)). If you choose a different registry, make sure the registry is OCI Image spec v1.1.0 compliant.

Reviewer: I think we need to emphasize that this registry is for testing purposes only.

### Building OCI-compatible registry image from source

> In some cases, such as if your development computer is running an Apple silicon processor, you can download the source of the image for the OCI-compatible registry. The following example clones the repository, builds the image using `docker buildx`, and creates a registry that is available at `localhost:5000`.

Reviewer: for the OCI-compatible registry => from the OCI-compatible registry

> In some cases, such as if your development computer is running an Apple silicon processor, you can download the source of the image for the OCI-compatible registry. The following example clones the repository, builds the image using `docker buildx`, and creates a registry that is available at `localhost:5000`.

Author: This is suggested by @zr-msft since `docker run -d -p 5000:5000 ghcr.io/oras-project/registry:v1.0.0-rc.3` doesn't work for Apple processor. Maybe we should ask distribution team to rebuild a new image?

> In some cases, such as if your development computer is running an Apple silicon processor, you can download the source of the image for the OCI-compatible registry. The following example clones the repository, builds the image using `docker buildx`, and creates a registry that is available at `localhost:5000`.

Reviewer 2: ideally, an additional image would be great, but building from source is a workaround we can ship near term

@FeynmanZhou this is a known issue: https://docs.docker.com/desktop/troubleshoot/known-issues/

```console
git clone https://github.com/oras-project/distribution.git
cd distribution

> docker buildx build -t oras-project/registry:v1.0.0-rc.3 . 

Reviewer: IMO, the build process is redundant here. It causes the Quick Start guide not to be quick and simple at all.

Author: I also think it could be too much for a quick start. We can remove this section once we know the reason why `registry:v1.0.0-rc.2` works but not `registry:v1.0.0-rc.3`.

Reviewer 2: @FeynmanZhou @yizha1 if a new image built for the apple silicon architecture can be created near-term, i agree we should pull this. If not, this needs to be included since the quickstart doesn't work for apple silicon macs as is.

Author: The distribution team is working on a new image that supports multi-arch. So two options: 
- Skip the `build registry` section for now, and add it back if needed
- Put `build registry` under the bottom of quick start as additional reference

docker push localhost:5000/net-monitor:v1
```

> Tags are mutable and a tag reference can point to a different container image than the one signed. Find out the digest value from the output of `docker push` command.

Reviewer 2: ```suggestion
Save the digest value of the image from the output of the `docker push` command.

*Important*: Always use the digest value of an image when signing since they are immutable. Tag values are mutable and can reference a different container image than the original signed container image.
```

An example output of `docker push`:

```output
The push refers to repository [localhost:5000/net-monitor]
2556c54bfdf3: Pushed
fb6ca4f9c8d3: Pushed
ded7a220bb05: Pushed
v1: digest: sha256:073b75987e95b89f187a89809f08a32033972bb63cda279db8a9ca16b7ff555a size: 942
```

> Use digest to identify the container image as `localhost:5000/net-monitor@sha256:073b75987e95b89f187a89809f08a32033972bb63cda279db8a9ca16b7ff555a`.

Reviewer 2: ```suggestion
In the above example, the reference to the container image using the digest value is `localhost:5000/net-monitor@sha256:073b75987e95b89f187a89809f08a32033972bb63cda279db8a9ca16b7ff555a`.
```

> Note：

> > Tags can be used to reference the container image. In this example, you can use tag `v1` for reference. Notation commands will resolve the tag `v1` to digest `sha256:073b75987e95b89f187a89809f08a32033972bb63cda279db8a9ca16b7ff555a` first. By validating the digest, you can make sure it is the right container image to reference.

Reviewer 2: ```suggestion
> If you use a tag value to sign an image, `notation` will determine the digest value of the current image associated and will use that digest to sign. Tags can be used to reference the container image. 
```

## List the signatures associated with the container image

Use `notation list` to show any signatures associated with the container image you built and pushed in the previous section.

```console

> notation list localhost:5000/net-monitor@sha256:073b75987e95b89f187a89809f08a32033972bb63cda279db8a9ca16b7ff555a

Reviewer 2: ```suggestion
IMAGE=localhost:5000/net-monitor@sha256:073b75987e95b89f187a89809f08a32033972bb63cda279db8a9ca16b7ff555a
notation list $IMAGE
```

Consider using the variable by setting it once and then using it throughout. It looks cleaner and is easier for users to copy/paste. With the current changes, they must update every command

docker push localhost:5000/net-monitor:v1
```

Save the digest value of the image from the output of the `docker push` command.

*Important*: Always use the digest value of an image when signing since they are immutable. Tag values are mutable and can reference a different container image than the original signed container image.

An example output of `docker push`:

```output
The push refers to repository [localhost:5000/net-monitor]
2556c54bfdf3: Pushed
fb6ca4f9c8d3: Pushed
ded7a220bb05: Pushed
v1: digest: sha256:073b75987e95b89f187a89809f08a32033972bb63cda279db8a9ca16b7ff555a size: 942
```

In the above example, the reference to the container image using the digest value is `localhost:5000/net-monitor@sha256:073b75987e95b89f187a89809f08a32033972bb63cda279db8a9ca16b7ff555a`.

> Note：
> If you use a tag value to sign an image, `notation` will determine the digest value of the current image associated and will use that digest to sign. Tags can be used to reference the container image.

## List the signatures associated with the container image

> Use `notation list` to show any signatures associated with the container image you built and pushed in the previous section.

Reviewer: ```suggestion
`notation ls`
```

```console
IMAGE=localhost:5000/net-monitor@sha256:073b75987e95b89f187a89809f08a32033972bb63cda279db8a9ca16b7ff555a

> notation list $IMAGE

Reviewer: ```suggestion
notation ls $IMAGE
```

Author: Do we need to follow the same rule as example to use alias for the command?

```

Confirm there are no signatures shown in the output.

## Generate a test key and self-signed certificate

> Use `notation certificate generate-test` to generate a test RSA key for signing artifacts, and a self-signed X.509 test certificate for verifying artifacts.

Reviewer: To keep consistency and simplicity, I would suggest using the short alias by default for all CLI commands in Notation docs and examples.

Author: @zr-msft @toddysm any thoughts on suggestion? I don't have strong opinions.

Reviewer 2: @FeynmanZhou @yizha1  as long as we stay consistent, I'm fine with short alias or full length
